src/v2/service.rs

```diff
@@ -72,8 +72,8 @@ pub struct Service {
 
     /// Environment variables and values to supply to the container.
     #[serde(default, skip_serializing_if = "BTreeMap::is_empty",
-            deserialize_with = "deserialize_map_or_key_value_list")]
-    pub environment: BTreeMap<String, RawOr<String>>,
+            deserialize_with = "deserialize_map_or_key_optional_value_list")]
```

Comment:
Have we looked at all the other callers of `deserialize_map_or_key_value_list` and verified that none of them require similar conversions? I can check this if you want, but it will probably have to wait until sometime next week at least.

Reply:
I tested all of them with `docker_compose` and the only one that *doesn't* accept valueless keys is `driver_opts`.  All the others do.

This raises some mildly interesting questions in cage, such as what to do if `io.fdy.cage.srcdir` is set but null.